cachi2/core/models/input.py

```diff
@@ -76,6 +76,30 @@ def _path_is_relative(cls, path: Path) -> Path:
         return check_sane_relpath(path)
 
 
+class _SSLOptions(pydantic.BaseModel, extra="forbid"):
+    """SSL options model.
+
+    Provides the 'ssl' key under 'options' for rpm package manager.
+
+    """
+
+    client_cert: Optional[str] = None
+    client_key: Optional[str] = None
+    ca_bundle: Optional[str] = None
+    ssl_verify: Optional[bool] = None
```

Comment:
I wonder what the implications are to allow users to easily disable the SSL cert/host verification. I know it emits a warning, but it can still be easily used in production envs. I do see its usefulness for testing, though.

If we decide that this is a big security issue, my suggestion is to scrap this option altogether, and think of a way to introduce it as a follow up (maybe by adding a warning in the resulting SBOM?).

Reply:
> If we decide that this is a big security issue, my suggestion is to scrap this option altogether, and think of a way to introduce it as a follow up (maybe by adding a warning in the resulting SBOM?).

This is not our problem to solve, those are user passthrough options that we can't take full responsibility for. Documentation and potentially a warning is all that we can really do about this. With these extra options the likelihood users asking for more options like this in the future to cover their exotic needs is IMO fairly decent, so they need to be prepared to carry the consequences (as long as we ourselves never disable it by default).

Reply:
I'd argue that we do take responsibility for some of these problems. For instance, we always require and validate checksums, this is not an user option.

I don't have strong feelings towards this, I don't think this is as grave as a checksum mismatch. If we think it is ok for this to be set in production environments, let's just keep the option here.